File: app/src/main/java/com/dgd/earthquakes/common/Presenter.java
package com.dgd.earthquakes.common;

import com.dgd.earthquakes.BaseApplication;
import com.dgd.earthquakes.data.IRepo;

/**
 * Created by max
 * on 04/10/17.
 */

public class Presenter<T> {

    protected final T view;

    public Presenter(T view){
        this.view = view;
    }

    protected IRepo getRepo() {
        return BaseApplication.getInstance().getRepository();
    }
}

File: app/src/main/java/com/dgd/earthquakes/common/Presenter.kt
package com.dgd.earthquakes.common

import com.dgd.earthquakes.BaseApplication
import com.dgd.earthquakes.data.IRepo

/**
 * Created by max
 * on 04/10/17.
 */

open class Presenter<T>(protected val view: T) : ScreenContract.IPresenter {

    protected val repo: IRepo
        get() = BaseApplication.instance?.repository!!
}
